### PR TITLE
Change all device operations to use add_instance_library 

### PR DESCRIPTION
**Pull request by cloudhan**
This will avoid many duplicated cmake configuration, and unify them.

### CODE REVIEW

**Review by asroy — DISMISSED** [review]

**Comment by cloudhan** [comment]
@asroy I think you need to merge it yourself, because it needs **write access** to proceed.

**Comment by asroy** [comment]
@cloudhan this PR is currently under CI, we will merge it once CI passed

**Review by asroy — APPROVED** [review]

**Pull request merged**
